packages/toolpad-core/src/createQuery.tsx

```diff
@@ -1,16 +1,27 @@
 import { TOOLPAD_QUERY } from './constants.js';
-import { PropValueType } from './types.js';
+import {
+  PrimitiveValueType,
+  NumberValueType,
+  BooleanValueType,
+  ArrayValueType,
+  ObjectValueType,
+  StringValueType,
+} from './types.js';
 
-export interface QueryResolverParams<P> {
-  parameters: P;
-}
-
-export interface QueryResolver<P> {
-  (params: QueryResolverParams<P>): Promise<any>;
-}
+type QueryParameterValue<V extends PrimitiveValueType> = V extends NumberValueType
```

Comment:
The drawback of this is that we have to keep it in sync with the `PrimitiveValueType`. Perhaps there is a way to do it based on the `default` property? Not sure if this works

```tsx
export interface ValueTypeBase<T = unknown> {
  type: 'string' | 'boolean' | 'number' | 'object' | 'array' | 'element' | 'template' | 'event';
  default?: T;
}

export interface StringValueType extends ValueTypeBase<string> {
  type: 'string';
  enum?: string[];
}

// ...

export type TsTypeOfPropValueType<T extends PropValueType> = NonNullable<T['default']>;
```